### PR TITLE
Replace Contacts picker by the Element iOS one (#638)

### PR DESCRIPTION
**Pull request by phlniji**
#638

### CODE REVIEW

**Comment by giomfo** [comment]
@phlniji After testing, I have the 2 remarks:
- we have to remove id reference from the search placeholder
image
We should consider the config flag define to hide or not the matrix id to add and use a new string. Note there are several string to update: `room_creation_invite_another_user` and `room_participants_invite_another_user`

- the room creation is broken, after the room type selection, the application displays the new contact picker and block the creation. We can remove the room members picker (like in Android)

**Comment by phlniji** [comment]
Those 2 points have been done in the last 2 commits

File: Riot/Modules/Contacts/DataSources/ContactsDataSource.m [hunk 1]
```diff
@@ -548,7 +548,8 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
         if (indexPath.row < filteredLocalContacts.count)
         {
             contact = filteredLocalContacts[indexPath.row];
-            showMatrixIdInDisplayName = YES;
+            // Tchap: Force to hide MatrixID in DisplayName
+//            showMatrixIdInDisplayName = YES;
```
Comment (giomfo) [inline Riot/Modules/Contacts/DataSources/ContactsDataSource.m:552]: The right code here in Element-iOS should be:
`showMatrixIdInDisplayName = self.forceMatrixIdInDisplayName`

I will discuss with you about a suggestion to improve forceMatrixIdInDisplayName property use in Element-iOS

File: Riot/Modules/Contacts/DataSources/ContactsDataSource.m [hunk 2]
```diff
@@ -557,7 +558,8 @@ - (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(N
         {
             contact = filteredMatrixContacts[indexPath.row];
             
-            showMatrixIdInDisplayName = self.forceMatrixIdInDisplayName ? YES : [isMultiUseNameByDisplayName[contact.displayName] isEqualToNumber:@(YES)];
+            // Tchap: Force to hide MatrixID in DisplayName
+//            showMatrixIdInDisplayName = self.forceMatrixIdInDisplayName ? YES : [isMultiUseNameByDisplayName[contact.displayName] isEqualToNumber:@(YES)];
```
Comment (giomfo) [inline Riot/Modules/Contacts/DataSources/ContactsDataSource.m:562]: I would keep the Element-iOS code here, normally the case where multiple accounts have the same display name is limited in Tchap. We can keep the code as soon as `forceMatrixIdInDisplayName` is `false`

File: Riot/Modules/Room/ParticipantsInviteModal/ContactsPicker/ContactsPickerViewModel.swift [hunk 3]
```diff
@@ -125,7 +125,11 @@ class ContactsPickerViewModel: NSObject, ContactsPickerViewModelProtocol {
         }
         
         contactsViewController.showSearch(true)
+#if HIDE_MATRIX_ID
```
Comment (giomfo) [inline Riot/Modules/Room/ParticipantsInviteModal/ContactsPicker/ContactsPickerViewModel.swift:128]: I don't want to add a new macro here, I would prefer to link the placeholder with the property `forceMatrixIdInDisplayName` defined just above
(let see our discussion to link this property with a new BuildSettings in Element)

File: Riot/Modules/StartChat/StartChatViewController.m [hunk 4]
```diff
@@ -114,8 +117,11 @@ - (void)viewDidLoad
     
     createBarButtonItem = [[UIBarButtonItem alloc] initWithTitle:[VectorL10n start] style:UIBarButtonItemStylePlain target:self action:@selector(onButtonPressed:)];
     self.navigationItem.rightBarButtonItem = createBarButtonItem;
-    
+#ifdef HIDE_MATRIX_ID
```
Comment (giomfo) [inline Riot/Modules/StartChat/StartChatViewController.m:120]: I don't want to add a new macro here, I would prefer to link the placeholder with the property `forceMatrixIdInDisplayName` defined just above
(let see our discussion to link this property with a new BuildSettings in Element)

File: Tchap/Modules/Room/Members/RoomParticipantsViewController.m [hunk 5]
```diff
@@ -26,9 +26,11 @@
 
 #import "MXCallManager.h"
 
+#import "ContactTableViewCell.h"
+
 #import "RageShakeManager.h"
 
-@interface RoomParticipantsViewController () <UITableViewDelegate, UITableViewDataSource, UISearchBarDelegate, UIGestureRecognizerDelegate, MXKRoomMemberDetailsViewControllerDelegate, ContactsViewControllerDelegate, RoomParticipantsInviteCoordinatorBridgePresenterDelegate>
+@interface RoomParticipantsViewController () <UITableViewDelegate, UITableViewDataSource, UISearchBarDelegate, UIGestureRecognizerDelegate, MXKRoomMemberDetailsViewControllerDelegate, RoomParticipantsInviteCoordinatorBridgePresenterDelegate>
```
Comment (giomfo) [inline Tchap/Modules/Room/Members/RoomParticipantsViewController.m:33]: Why do we still use a specific Tchap `RoomParticipantsViewController` class? (I think we can switch to the Element one)

**Review by giomfo — COMMENTED** [review]
commented

**Comment by giomfo** [comment]
one more comment in the room creation screen the top right action must be "Create" instead of "Next"
image

File: Tchap/Modules/Room/Members/RoomParticipantsViewController.m (hunk 5 above)
Comment (phlniji) [reply Tchap/Modules/Room/Members/RoomParticipantsViewController.m:33]: This is removed in next PR (code cleaning).

**Review by giomfo — APPROVED** [review]
LGTM

**Pull request merged**